frontend/src/component/ui/Modal.spec.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { Modal } from './Modal';
+
+jest.useFakeTimers();
+
+it('renders without crashing', () => {
+  render(<Modal />);
+});
+
+it('opens correctly', () => {
+  const modalRef = React.createRef<Modal>();
+  render(<Modal ref={modalRef} canClose={true} />);
```

Comment:
```suggestion
  render(<Modal ref={modalRef} canClose={true}>Modal content</Modal>);
```

Reply:
Let's check whether the content of the modal is present to determine whether it's open or closed.

Currently, we are checking the internal state of the modal to verify its status. This means that every time when we change the Modal's internal implementation, we may have to update the test correspondingly even though the consumer of the modal is not affected. This incurs unnecessary  maintenance overhead.